File: gamslib/prodplan/prodplan-scalar.py
#  MIP written by GAMS Convert at 12/13/18 11:30:37
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         17        9        0        8        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         25       17        8        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         64       64        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x2 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x3 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x4 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x5 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x6 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x7 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x8 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x9 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x10 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x11 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x12 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x13 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x14 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x15 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x16 = Var(within=Reals,bounds=(0,None),initialize=0)
m.b17 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b18 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b19 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b20 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b21 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b22 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b23 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b24 = Var(within=Binary,bounds=(0,1),initialize=0)

m.obj = Objective(expr=   5*m.x1 + 5*m.x2 + 5*m.x3 + 5*m.x4 + 5*m.x5 + 5*m.x6 + 5*m.x7 + 2.5*m.x8 + 100*m.x9 + 100*m.x10
                        + 100*m.x11 + 100*m.x12 + 100*m.x13 + 100*m.x14 + 100*m.x15 + 100*m.x16 + 5000*m.b17
                        + 5000*m.b18 + 5000*m.b19 + 5000*m.b20 + 5000*m.b21 + 5000*m.b22 + 5000*m.b23 + 5000*m.b24
                       , sense=minimize)

m.c2 = Constraint(expr=   m.x9 - 5600*m.b17 <= 0)

m.c3 = Constraint(expr=   m.x10 - 6800*m.b18 <= 0)

m.c4 = Constraint(expr=   m.x11 - 6400*m.b19 <= 0)

m.c5 = Constraint(expr=   m.x12 - 5600*m.b20 <= 0)

m.c6 = Constraint(expr=   m.x13 - 4800*m.b21 <= 0)

m.c7 = Constraint(expr=   m.x14 - 3600*m.b22 <= 0)

m.c8 = Constraint(expr=   m.x15 - 2400*m.b23 <= 0)

m.c9 = Constraint(expr=   m.x16 - 1200*m.b24 <= 0)

m.c10 = Constraint(expr= - m.x1 + m.x9 == 200)

m.c11 = Constraint(expr=   m.x1 - m.x2 + m.x10 == 400)

m.c12 = Constraint(expr=   m.x2 - m.x3 + m.x11 == 800)

m.c13 = Constraint(expr=   m.x3 - m.x4 + m.x12 == 800)

m.c14 = Constraint(expr=   m.x4 - m.x5 + m.x13 == 1200)

m.c15 = Constraint(expr=   m.x5 - m.x6 + m.x14 == 1200)

m.c16 = Constraint(expr=   m.x6 - m.x7 + m.x15 == 1200)

m.c17 = Constraint(expr=   m.x7 - m.x8 + m.x16 == 1200)

File: gamslib/prodplan/prodplan-scalar.jl
#  MIP written by GAMS Convert at 10/11/20 13:33:52
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         17        9        0        8        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         25       17        8        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         64       64        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1, start=0)
@variable(m, 0 <= x2, start=0)
@variable(m, 0 <= x3, start=0)
@variable(m, 0 <= x4, start=0)
@variable(m, 0 <= x5, start=0)
@variable(m, 0 <= x6, start=0)
@variable(m, 0 <= x7, start=0)
@variable(m, 0 <= x8, start=0)
@variable(m, 0 <= x9, start=0)
@variable(m, 0 <= x10, start=0)
@variable(m, 0 <= x11, start=0)
@variable(m, 0 <= x12, start=0)
@variable(m, 0 <= x13, start=0)
@variable(m, 0 <= x14, start=0)
@variable(m, 0 <= x15, start=0)
@variable(m, 0 <= x16, start=0)
@variable(m, 0 <= b17 <= 1, binary=true, start=0)
@variable(m, 0 <= b18 <= 1, binary=true, start=0)
@variable(m, 0 <= b19 <= 1, binary=true, start=0)
@variable(m, 0 <= b20 <= 1, binary=true, start=0)
@variable(m, 0 <= b21 <= 1, binary=true, start=0)
@variable(m, 0 <= b22 <= 1, binary=true, start=0)
@variable(m, 0 <= b23 <= 1, binary=true, start=0)
@variable(m, 0 <= b24 <= 1, binary=true, start=0)

@objective(m, Min, 5*x1 + 5*x2 + 5*x3 + 5*x4 + 5*x5 + 5*x6 + 5*x7 + 2.5*x8 + 100*x9 + 100*x10 + 100*x11 + 100*x12
     + 100*x13 + 100*x14 + 100*x15 + 100*x16 + 5000*b17 + 5000*b18 + 5000*b19 + 5000*b20 + 5000*b21 + 5000*b22
     + 5000*b23 + 5000*b24)

@constraint(m, x9 - 5600*b17 <= 0)

@constraint(m, x10 - 6800*b18 <= 0)

@constraint(m, x11 - 6400*b19 <= 0)

@constraint(m, x12 - 5600*b20 <= 0)

@constraint(m, x13 - 4800*b21 <= 0)

@constraint(m, x14 - 3600*b22 <= 0)

@constraint(m, x15 - 2400*b23 <= 0)

@constraint(m, x16 - 1200*b24 <= 0)

@constraint(m,  - x1 + x9 == 200)

@constraint(m, x1 - x2 + x10 == 400)

@constraint(m, x2 - x3 + x11 == 800)

@constraint(m, x3 - x4 + x12 == 800)

@constraint(m, x4 - x5 + x13 == 1200)

@constraint(m, x5 - x6 + x14 == 1200)

@constraint(m, x6 - x7 + x15 == 1200)

@constraint(m, x7 - x8 + x16 == 1200)
